src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleNotificationHandler.java

```diff
@@ -0,0 +1,42 @@
+package org.radarbase.appserver.service.questionnaire.protocol;
+
+import org.radarbase.appserver.dto.protocol.Assessment;
+import org.radarbase.appserver.dto.questionnaire.AssessmentSchedule;
+import org.radarbase.appserver.entity.Notification;
+import org.radarbase.appserver.entity.Task;
+import org.radarbase.appserver.entity.User;
+import org.radarbase.appserver.service.FcmNotificationService;
+import org.radarbase.appserver.service.questionnaire.notification.TaskNotificationGeneratorService;
+import org.radarbase.appserver.service.questionnaire.notification.NotificationType;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SimpleNotificationHandler implements ProtocolHandler {
+    private transient TaskNotificationGeneratorService taskNotificationGeneratorService = new TaskNotificationGeneratorService();
+    private transient FcmNotificationService notificationService;
+
+    public SimpleNotificationHandler(FcmNotificationService notificationService) {
+        this.notificationService = notificationService;
+    }
+
+    @Override
+    public AssessmentSchedule handle(AssessmentSchedule assessmentSchedule, Assessment assessment, User user) {
+        List<Notification> notifications = generateNotifications(assessmentSchedule.getTasks(), user);
+        this.notificationService.addNotifications(notifications, user);
+        return assessmentSchedule;
+    }
+
+    public List<Notification> generateNotifications(List<Task> tasks, User user) {
+        return tasks.parallelStream()
+                .map(task ->{
+                        Notification notification = this.taskNotificationGeneratorService
+                                .createNotification(task, NotificationType.NOW, task.getTimestamp().toInstant());
+                        notification.setUser(user);
+                        return notification;
+                })
+                .filter(notification-> (Instant.now().isBefore(notification.getScheduledTime()))).collect(Collectors.toList());
```

Comment:
Should we only filter based on scheduledTime + CompletionWindow ? Because the task is still valid and hence worth sending a notification for.

Reply:
Ah yes, I've updated it to use `ttlSeconds` now, which is calculated from the completion window.